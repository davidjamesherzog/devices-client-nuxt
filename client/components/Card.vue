<template>
  <div class="column">
    <div class="card">
      <header class="card-header">
        <p class="card-header-title has-text-grey" data-card-title>
          {{ title }}
        </p>
      </header>
      <div class="card-content">
        <div class="content has-text-centered">
          <b-icon
            :icon="icon"
            size="is-large"
            type="is-primary"
            data-card-icon
          />
        </div>
      </div>
      <footer class="card-footer">
        <div class="card-footer-item">
          <span>
            <slot />
          </span>
        </div>
      </footer>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'nuxt-property-decorator';

@Component({
  name: 'Card'
})
export default class Card extends Vue {
  @Prop({ type: String })
  private title!: string;

  @Prop({ type: String })
  private icon!: string;
}
</script>
