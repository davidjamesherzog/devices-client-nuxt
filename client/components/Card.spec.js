import { createLocalVue, mount } from '@vue/test-utils';
import Buefy from 'buefy';
import Card from '~/components/Card.vue';

describe('Card', () => {
  let localVue;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Buefy);
  });

  test('is a Vue instance', () => {
    const wrapper = mount(Card, {
      localVue
    });
    expect(wrapper.vm).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  test('should render title', () => {
    const wrapper = mount(Card, {
      localVue,
      propsData: {
        title: 'My Card'
      }
    });
    const title = wrapper.find('[data-card-title]');
    expect(title.exists()).toBeTruthy();
    expect(title.text()).toEqual('My Card');
    expect(wrapper).toMatchSnapshot();
  });

  test('should render icon', () => {
    const wrapper = mount(Card, {
      localVue,
      propsData: {
        icon: 'cellphone-link'
      }
    });
    const icon = wrapper.find('[data-card-icon]');
    expect(icon.exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
