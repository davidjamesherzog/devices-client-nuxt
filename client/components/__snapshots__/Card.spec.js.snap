// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card is a Vue instance 1`] = `
<div class="column">
  <div class="card">
    <header class="card-header">
      <p data-card-title="" class="card-header-title has-text-grey">

      </p>
    </header>
    <div class="card-content">
      <div class="content has-text-centered"><span class="icon has-text-primary is-large" data-card-icon=""><i class="mdi mdi-undefined mdi-48px"></i></span></div>
    </div>
    <footer class="card-footer">
      <div class="card-footer-item"><span></span></div>
    </footer>
  </div>
</div>
`;

exports[`Card should render icon 1`] = `
<div class="column">
  <div class="card">
    <header class="card-header">
      <p data-card-title="" class="card-header-title has-text-grey">

      </p>
    </header>
    <div class="card-content">
      <div class="content has-text-centered"><span class="icon has-text-primary is-large" data-card-icon=""><i class="mdi mdi-cellphone-link mdi-48px"></i></span></div>
    </div>
    <footer class="card-footer">
      <div class="card-footer-item"><span></span></div>
    </footer>
  </div>
</div>
`;

exports[`Card should render title 1`] = `
<div class="column">
  <div class="card">
    <header class="card-header">
      <p data-card-title="" class="card-header-title has-text-grey">
        My Card
      </p>
    </header>
    <div class="card-content">
      <div class="content has-text-centered"><span class="icon has-text-primary is-large" data-card-icon=""><i class="mdi mdi-undefined mdi-48px"></i></span></div>
    </div>
    <footer class="card-footer">
      <div class="card-footer-item"><span></span></div>
    </footer>
  </div>
</div>
`;
